src/encoded/static/components/collection.js

```diff
@@ -169,21 +169,18 @@ var lookup_column = function (result, column) {
         },
 
         guessColumns: function (props) {
-            var column_list = props.columns || props.context.columns;
             var columns = [];
-            if (!column_list || Object.keys(column_list).length === 0) {
-                for (var key in props.context['@graph'][0]) {
-                    if (key.slice(0, 1) != '@' && key.search(/(uuid|_no|accession)/) == -1) {
+            // Get columns right from results. Selective embedding defines cols used
+            for (var key in props.context['@graph'][0]) {
+                if (key.slice(0, 1) != '@' && key.search(/(uuid|_no|accession)/) == -1) {
```

Comment:
probably a comment about leaving out special fields here would be helpful for the next guy that has to debug this stuff..